backend/src/app.py

```diff
@@ -77,48 +77,109 @@ def add_user():
 
             with Session.begin() as session:
-                user_query = session.query(User).filter_by(username=username)
-                if session.query(user_query.exists()).scalar():
+                if user_exists(session, username):
                     return (
                         jsonify(
-                            {"success": True, "user": user_query.first().user_info()}
+                            {
+                                "success": True,
+                                "user": get_user_from_db(session, username).user_dict(),
+                            }
                         ),
                         200,
                     )
             return jsonify({"success": False, "msg": "User does not exist"}), 404
+
         elif request.method == "DELETE":
             username = request.args["username"]
 
             # Error check
-            if username == None or username == "":
+            if username is None or username == "":
                 return jsonify({"response": "username cannot be blank"}), 400
 
             with Session.begin() as session:
-                user_query = session.query(User).filter_by(username=username)
-                if session.query(user_query.exists()).scalar():
-                    user = user_query.first()
+                if user_exists(session, username):
+                    user = get_user_from_db(session, username)
                     session.delete(user)
                     return (
                         jsonify({"success": True, "msg": "User has been deleted"}),
                         200,
                     )
             return jsonify({"success": False, "msg": "User does not exist"}), 204
 
-    @app.route("/updateuser", methods=["POST"])
-    def update_user():
-        return ""
+    ## Account endpoints
+    @app.route("/account", methods=["POST"])
+    def add_account():
+        if request.method == "POST":
+            account_info = request.json
 
-    @app.route("/deleteuser", methods=["GET"])
-    def delete_user():
-        return ""
+            # Error check
+            if account_info["username"] is None or account_info["username"] == "":
+                return jsonify({"response": "username cannot be blank"}), 400
+            if not account_type_valid(account_info["account_type"]):
+                return jsonify({"response": "Account type is not valid"}), 400
 
-    @app.route("/viewuser", methods=["GET"])
-    def view_user():
-        return ""
+            with Session.begin() as session:
+                if not user_exists(session, account_info["username"]):
+                    return (
+                        jsonify({"success": False, "msg": "User does not exist"}),
+                        204,
+                    )
+                elif (
+                    account_info["account_id"] == ""
+                    or account_info["account_id"] is None
+                ):
```

Comment:
May want to add a comment describing that account creation is indicated by an "invalid" input in account and that the frontend will be responsible for ensuring this field is set to what it is not just as a mistake.